File: atcoder.jp/contests/abc064/tasks/abc064_b/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];

const max = <T extends number | bigint>(array: T[]) =>
  array.reduce((a, b) => (a < b ? b : a));
const min = <T extends number | bigint>(array: T[]) =>
  array.reduce((a, b) => (a < b ? a : b));

input();
const A = input().split(" ").map(Number);

const ans = max(A) - min(A);

console.log(ans);

File: atcoder.jp/contests/abc064/tasks/abc064_b/main.go
package main

import (
	"fmt"
)

func InputInt() int {
	var n int
	fmt.Scanf("%d", &n)
	return n
}

func InputIntArray(n int) []int {
	result := make([]int, n)
	for i := range result {
		fmt.Scanf("%d", &result[i])
	}
	return result
}

func main() {
	N := InputInt()
	A := InputIntArray(N)

	max := 0
	min := 10000

	for _, v := range A {
		if max < v {
			max = v
		}
		if v < min {
			min = v
		}
	}

	ans := max - min

	fmt.Println(ans)
}
